src/utils/lightcurves.rs

```diff
@@ -314,3 +316,388 @@ fn linear_fit(time: Vec<f32>, mag: Vec<f32>, mag_err: Vec<f32>) -> Option<Linear
         nb_data: n as i32,
     })
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    #[tokio::test]
+    async fn test_flux2mag() {
```

Comment:
None of the tests in this module need to be async. Can write this instead:

```suggestion
    #[test]
    fn test_flux2mag() {
```

And so forth.

Reply:
removed